enterprise/backend/test/metabase_enterprise/serialization/v2/load_test.clj

```diff
@@ -343,3 +343,262 @@
                                  :aggregation  [[:count]]}
                       :database (:id @db1d)}
                      (:dataset_query @card1d))))))))))
+
+(deftest segment-test
+  ;; Segment.definition is a JSON-encoded MBQL query, which contain database, table, and field IDs - these need to be
+  ;; converted to a portable form and read back in.
+  ;; This test has a database, table and fields, that exist on both sides with different IDs, and expects a segment that
+  ;; references those fields to be correctly loaded with the dest IDs.
+  (testing "embedded MBQL in Segment :definition is portable"
+    (let [serialized (atom nil)
+          coll1s     (atom nil)
+          db1s       (atom nil)
+          table1s    (atom nil)
+          field1s    (atom nil)
+          seg1s      (atom nil)
+          user1s     (atom nil)
+          db1d       (atom nil)
+          table1d    (atom nil)
+          field1d    (atom nil)
+          user1d     (atom nil)
+          seg1d      (atom nil)
+          db2d       (atom nil)
+          table2d    (atom nil)
+          field2d    (atom nil)]
+
+
+      (ts/with-source-and-dest-dbs
+        (testing "serializing the original database, table, field and card"
+          (ts/with-source-db
+            (reset! coll1s  (ts/create! Collection :name "pop! minis"))
```

Comment:
any reasons why you don't use `mt/with-temp*` for creating these records?

Reply:
Because they need to be accessed by both the `with-source-db` and `with-dest-db` sections, but can only be created inside either of them.
I suppose I could do
```clojure
(let [... atoms]
  (mt/with-source-db
    (mt/with-temp [Collection {... :as c1} {...}]
      (reset! c1-atom c1))))
```
but it doesn't seem like a significant improvement.